src/python/director/motionplanningpanel.py

```diff
@@ -0,0 +1,391 @@
+        om.removeFromObjectModel(om.findObjectByName('footstep plan'))
+        om.removeFromObjectModel(om.findObjectByName('iDRMStancePoses'))
+        
+    @staticmethod
+    def getConstraintFrameFolder():
+        return om.getOrCreateContainer('constraint frames', parentObj=om.getOrCreateContainer('teleop plan', parentObj=om.findObjectByName('planning')))
+    
+    def removeHandFrames(self):
+        linkName = self.ikPlanner.getHandLink('left')
+        frameName = '%s constraint frame' % linkName
+        om.removeFromObjectModel(om.findObjectByName(frameName))
+        linkName = self.ikPlanner.getHandLink('right')
+        frameName = '%s constraint frame' % linkName
+        om.removeFromObjectModel(om.findObjectByName(frameName))
+        
+    def createHandGoalFrame(self):
+        if self.getReachHand() == 'Left':
+            side = 'left'
+        elif self.getReachHand() == 'Right':
+            side = 'right'
+        else:
+            side = 'none'
+        self.removeHandFrames()
+        
+        if not side == 'none':
+            folder = self.getConstraintFrameFolder()
+            startPose = np.array(self.robotStateJointController.q)
+            linkName = self.ikPlanner.getHandLink(side)
+            frameName = '%s constraint frame' % linkName
+            graspToHand = self.ikPlanner.newPalmOffsetGraspToHandFrame(side, self.palmOffsetDistance)
+            graspToWorld = self.ikPlanner.newGraspToWorldFrame(startPose, side, graspToHand)
+            frame = vis.showFrame(graspToWorld, frameName, parent=folder, scale=0.2)
+            frame.connectFrameModified(self.onGoalFrameModified)
+        
+    def updateIk(self):
+        if not self.constraintSet:
+            self.updateIKConstraints()
+        endPose, info = self.constraintSet.runIk()
+        endPoseName = 'reach_end'
+        self.ikPlanner.addPose(endPose, endPoseName)
+        self.showPose(self.constraintSet.endPose)
+        app.displaySnoptInfo(info)
+        
+        if self.ui.walkingPlanningPanel.enabled and self.ui.walkInteractiveCheck.checked:
+            self.onWalkingPlan()
+            
+    def onSearchFinalPose(self):
+        self.updateIk()
+        
+    def onGoalFrameModified(self, frame):
+        if self.ui.fpInteractiveCheck.checked:
+            self.updateIk()
+            
+    def showPose(self, pose):
+        self.teleopJointController.setPose('MP_EndPose', pose)
+        self.hidePlanFunction()
+        self.showMPModel()
+        
+    def showMPModel(self):
+        self.teleopRobotModel.setProperty('Visible', True)
+        self.robotStateModel.setProperty('Visible', True)
+        self.robotStateModel.setProperty('Alpha', 0.1)
+        
+        
+    def getCurrentWalkingGoal(self):
+        t = self.footDriver.getFeetMidPoint(self.teleopRobotModel)
+        t.PreMultiply()
+        t.Translate(0.0, 0.0, 0.0)
+        t.PostMultiply()
+        return t
+    
+    def onWalkingPlan(self):
```

Comment:
it seems like this function has a lot of copy and pasted code from the onNewWalkingGoal() function in footstepsdriverpanel.py.  Perhaps the common code could be placed into footstepsdriver.py so that it may be reused?